azure/iac/Redis.rego

```diff
@@ -142,4 +142,143 @@ serverRole_metadata := {
     "Resource Type": "Microsoft.Cache/redis/linkedservers",
     "Policy Help URL": "",
     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis/linkedservers"
+}
+
+
+
+# PR-AZR-ARM-ARC-004
+
+default redis_public_access = null
+
+
+azure_attribute_absence["redis_public_access"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.publicNetworkAccess
+}
+
+source_path[{"redis_public_access":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.publicNetworkAccess
+    metadata:= {
+        "resource_path": [["resources",i,"properties","publicNetworkAccess"]]
+    }
+}
+
+azure_issue["redis_public_access"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    lower(resource.properties.publicNetworkAccess) != "disabled"
+}
+
+source_path[{"redis_public_access":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.cache/redis"
+    lower(resource.properties.publicNetworkAccess) != "disabled"
+    metadata:= {
+        "resource_path": [["resources",i,"properties","publicNetworkAccess"]]
+    }
+}
+
+
+redis_public_access {
+    azure_attribute_absence["redis_public_access"]
```

Comment:
@shahinshirvani Default value is `'Enabled'` if `publicNetworkAccess `does not exist.
Ref: https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis?tabs=json#rediscreateproperties

Reply:
fixed